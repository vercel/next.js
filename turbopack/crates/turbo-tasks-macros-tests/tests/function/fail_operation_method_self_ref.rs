#![allow(dead_code)]
#![feature(arbitrary_self_types)]
#![feature(arbitrary_self_types_pointers)]

use turbo_tasks::Vc;

#[turbo_tasks::value]
struct Foobar;

#[turbo_tasks::value_impl]
impl Foobar {
    #[turbo_tasks::function(operation)]
    fn self_ref(&self) -> Vc<()> {
        Vc::cell(())
    }
}

fn main() {}
