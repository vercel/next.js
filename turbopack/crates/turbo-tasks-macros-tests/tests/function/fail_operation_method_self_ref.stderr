error: methods taking `self` are not supported with `operation`
  --> tests/function/fail_operation_method_self_ref.rs:13:17
   |
13 |     fn self_ref(&self) -> Vc<()> {
   |                 ^^^^^
